File: create_test_pdf.py
from reportlab.lib.pagesizes import letter
from reportlab.pdfgen import canvas
import os

c = canvas.Canvas('test.pdf', pagesize=letter)
w, h = letter

c.setFont('Helvetica-Bold', 24)
c.drawCentredString(w/2, h-100, 'Nexino PrintFlow')
c.setFont('Helvetica', 14)
c.drawCentredString(w/2, h-130, 'Test Document - Page 1')
c.setFont('Helvetica', 12)
c.drawString(72, h-200, 'This is a test PDF document for Nexino PrintFlow.')
c.drawString(72, h-220, 'It simulates a real print job uploaded by a customer.')
c.drawString(72, h-240, 'Features tested:')
c.drawString(90, h-260, '- PDF upload and validation')
c.drawString(90, h-280, '- Page count detection')
c.drawString(90, h-300, '- Price calculation')
c.drawString(90, h-320, '- Job tracking')
c.showPage()

c.setFont('Helvetica-Bold', 18)
c.drawCentredString(w/2, h-100, 'Page 2')
c.setFont('Helvetica', 12)
c.drawString(72, h-150, 'This page demonstrates multi-page support.')
c.drawString(72, h-170, 'The system counts all pages for accurate pricing.')
c.showPage()

c.setFont('Helvetica-Bold', 18)
c.drawCentredString(w/2, h-100, 'Page 3')
c.setFont('Helvetica', 12)
c.drawString(72, h-150, 'Final page of the test document.')
c.showPage()

c.save()
size = os.path.getsize('test.pdf')
print(f'Created test.pdf ({size} bytes)')

File: create-test-pdf.mjs
import { PDFDocument } from 'pdf-lib';
import fs from 'fs';

async function createTestPdf() {
  const doc = await PDFDocument.create();
  
  const page1 = doc.addPage([612, 792]);
  page1.drawText('Nexino PrintFlow', { x: 180, y: 700, size: 24 });
  page1.drawText('Test Document - Page 1', { x: 200, y: 660, size: 14 });
  page1.drawText('This is a test PDF for Nexino PrintFlow.', { x: 72, y: 600, size: 12 });
  page1.drawText('Features: upload, validation, pricing, job tracking.', { x: 72, y: 580, size: 12 });

  const page2 = doc.addPage([612, 792]);
  page2.drawText('Page 2', { x: 260, y: 700, size: 18 });
  page2.drawText('Multi-page support for print jobs.', { x: 72, y: 650, size: 12 });

  const page3 = doc.addPage([612, 792]);
  page3.drawText('Page 3', { x: 260, y: 700, size: 18 });
  page3.drawText('Final page of test document.', { x: 72, y: 650, size: 12 });

  const pdfBytes = await doc.save();
  fs.writeFileSync('test.pdf', pdfBytes);
  console.log(`Created test.pdf (${pdfBytes.length} bytes)`);
}

createTestPdf().catch(console.error);
